GetStream/stream-chat-swift
# Mark as unread

### 🔗 Issue Links

https://github.com/GetStream/ios-issues-tracking/issues/285

### 🎯 Goal

This PR adds an option to mark as unread a chat starting from a particular message

### 📝 Summary

Now it is possible to select a particular message, and mark the channel as unread starting from that message.

### 🛠 Implementation

ChannelController exposes the first unread message id, and this one is used in the UI to show a banner (not part of this PR) to visually distinguish new messages

### 🧪 Manual Testing Notes

1. Open a channel
2. Long press on a message
3. Select "Mark As Unread"

Expected result:
- The channel should be marked as unread starting from that message

### ☑️ Contributor Checklist


### 🎁 Meme

![](https://media.giphy.com/media/fxHwi5ia9NgTCrl92Z/giphy.gif)
